### PR TITLE
docs improvement

### PR DESCRIPTION
**Pull request by smakalias**
## Summary of the changes / Why this improves CrateDB

related to #8541 
I believe it's an improvement to the existing section

## Checklist

 - [ ] User relevant changes are recorded in ``CHANGES.txt``
 - [ ] Touched code is covered by tests
 - [x] Documentation has been updated if necessary
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: blackbox/docs/general/builtins/table-functions.rst [hunk 1]
```diff
@@ -34,17 +35,31 @@ values will be returned for the functions that are exhausted. An example::
...
 
 .. contents::
    :local:
 
+.. _table-functions-scalar:
+
+Scalar functions
+================
+
+:ref:`scalar`, when used in the ``FROM`` clause of a query as a table or subquery,
```
Comment (mfussenegger) [inline blackbox/docs/general/builtins/table-functions.rst]: What are you referring to with `as a table or subquery` ? Not sure if this makes more sense to list all types of possible relations and imply that the scalar can be used as such. I think the point is that the scalar can be used in the FROM and that it will act like any other relation would.

Comment (smakalias) [reply blackbox/docs/general/builtins/table-functions.rst]: > What are you referring to with as a table or subquery?
The proper term would be to use the `relation` abstraction. And, I agree that enumerating all different options is not nice, however, it would be inconsistent with other places I saw. I'm happy to change it there as well though and keep the relation abstraction.
 
> I think the point is that the scalar can be used in the FROM and that it will act like any other relation would.

The key point (and what I wanted to stress here) is the "like any other relation would").
I think it's an important point as a function can be used in the `From` clause as an expression and not as a relation eg in 
`from substr(initcap('hello world'),1, 5)` 
`substr` scalar will be treated as a relation while `initcap` wouldn't despite the fact that both are used in the query `From` clause.

My point was to highlight this difference.

File: blackbox/docs/appendices/release-notes/unreleased.rst [hunk 2]
```diff
@@ -145,7 +145,7 @@ Changes
 =======
 
 - Enabled Scalar function evaluation when used :ref:`in the query FROM
-  clause <table-functions-scalar>`.
+  clause as a table or subquery<table-functions-scalar>`.
```
Comment (mfussenegger) [inline blackbox/docs/appendices/release-notes/unreleased.rst]:
```suggestion
  clause in place of a relation <table-functions-scalar>`.
```

File: blackbox/docs/general/builtins/table-functions.rst [hunk 3]
```diff
@@ -6,8 +6,9 @@
 Table Functions
 ===============
 
-Table functions are functions that produce a set of rows. They are used like a
-table or subquery in the ``FROM`` clause of a query.
+Table functions are functions that produce a set of rows which are made up of
+either base data types (scalar types) or composite data types (table rows).
```
Comment (mfussenegger) [inline blackbox/docs/general/builtins/table-functions.rst]: I'd revert this change. I don't understand this "either base data types (scalar types) or composite data types addition. I think this confuses more than it clarifies.

File: blackbox/docs/general/builtins/table-functions.rst [hunk 4]
```diff
@@ -6,8 +6,9 @@
 Table Functions
 ===============
 
-Table functions are functions that produce a set of rows. They are used like a
-table or subquery in the ``FROM`` clause of a query.
+Table functions are functions that produce a set of rows which are made up of
```
Comment (mfussenegger) [inline blackbox/docs/general/builtins/table-functions.rst]:
```suggestion
Table functions are functions that produce a set of rows. They can either be used in place of a relation in the FROM clause, or within the select list of a query.

If used within the select list, the table functions will be evaluated per row ...
```

File: blackbox/docs/general/builtins/table-functions.rst (hunk 1 above)
Comment (mfussenegger) [inline blackbox/docs/general/builtins/table-functions.rst]:
```suggestion
:ref:`scalar`, when used in the ``FROM`` clause in place of a relation, 
```
